src/components/Page.js

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { getData } from '../apiCalls';
+import Rule from './Rule';
+import Ability from './Ability';
+import Skill from './Skill';
+import Class from './Class';
+import Race from './Race';
+
+const Page = ({setError}) => {
+  const location = useLocation();
+  const URL1 = location.pathname.split('/')[1];
+  const URL2 = location.pathname.split('/')[2]
+  const [page, setPage] = useState({});
+  const [loading, setLoading] = useState(true)
+
+  const PATHS = {
+    'gameplay-basics': 'rule-sections',
+    'abilities': 'ability-scores',
+    'skills': 'skills',
+    'races': 'races',
+    'classes': 'classes'
+  }
+
+  useEffect(() => {
+    PATHS[URL1] === 'ability-scores' ? 
+      getData(PATHS[URL1], URL2.slice(0, 3))
+      .then(res => setPage(res))
+      .then(() => setLoading(false))
+      .catch(err => setError(err))
+    :
+      getData(PATHS[URL1], URL2)
+      .then(res => setPage(res))
+      .then(() => setLoading(false)) 
+      .catch(err => setError(err))  
+  }, [])
+
```

Comment:
This is great! Love that getData just gets invoked on page load of each `Page`

I think that we could avoid repeating some code still be changing this to something like: 
```
useEffect(() => {
const secondUrl = PATHS[URL1] === 'ability-scores' ? URL2.slice(0, 3) : URL2
      getData(PATHS[URL1], secondUrl)
      .then(res => setPage(res))
      .then(() => setLoading(false))
      .catch(err => setError(err))  
  }, [])
```

Reply:
Thank you!! I tried this a couple of different ways, but wasn't getting it right, so I put it on the back burner. This is exactly what I needed! You're the best.

Reply:
done